test/gtest/ternary_tensor_ops.cpp

```diff
@@ -0,0 +1,379 @@
+                                                   {20, 16, 8},
+                                                   {1, 16, 8},
+                                                   {16, 8},
+                                                   {8}};
+
+std::vector<std::vector<size_t>> tensorBLensArr = {{32, 16, 8, 4, 4}, // tensor B
+                                                   {32, 16, 1, 1, 1},
+                                                   {1, 16, 8, 1, 1},
+                                                   {1, 1, 8, 4, 1},
+                                                   {16, 20, 16, 8},
+                                                   {16, 20, 16, 1},
+                                                   {16, 20, 1, 1},
+                                                   {16, 1, 1, 1},
+                                                   {1, 20, 16, 8},
+                                                   {1, 20, 16, 1},
+                                                   {1, 20, 1, 1},
+                                                   {1, 1, 16, 8},
+                                                   {1, 1, 1, 8},
+                                                   {20, 16, 8},
+                                                   {20, 16, 1},
+                                                   {1, 16, 8},
+                                                   {1, 16, 1},
+                                                   {20, 1, 1},
+                                                   {16, 8},
+                                                   {16, 1},
+                                                   {1, 8},
+                                                   {8},
+                                                   {1}};
+
+std::vector<std::vector<int64_t>> offsetsArr = {
+    {0, 0, 0}, {64, 32, 16}, {32, 16, 32}, {32, 16, 32}};
+
+std::vector<std::vector<float>> alphabetaArr = {{1, 1, 0}, {-1, 1, 1}, {1.0, 0.5, 0.3}};
+
+std::vector<std::vector<size_t>> stridesArr = {{8 * 16 * 20 * 16, 8 * 16 * 20, 8 * 16, 8, 1}};
+
+std::vector<bool> packedArr = {true, false};
+
+std::vector<miopenTensorOp_t> operationArr = {
+    miopenTensorOpAdd, miopenTensorOpMul, miopenTensorOpMin, miopenTensorOpMax};
+} // namespace
+
+struct TestCase
+{
+    std::vector<size_t> tensorlens_ac;
+    std::vector<size_t> tensorlens_b;
+    std::vector<int64_t> offsets;
+    std::vector<size_t> stride_a;
+    std::vector<size_t> stride_b;
+    std::vector<size_t> stride_c;
+    std::vector<float> alphabeta;
+    bool packed;
+    miopenTensorOp_t operation;
+};
+
+template <typename T>
+struct TensorOpsCommon : public testing::TestWithParam<TestCase>
+{
+    void SetUp() override
+    {
+        prng::reset_seed();
+
+        CreateTensors();
+
+        std::vector<T> tensorGPUData = CalculateOnGPU();
+        std::vector<T> tensorCPUData = CalculateOnCPU();
+
+        CompareResults(tensorGPUData, tensorCPUData);
```

Comment:
This is the body of the test, it's better to place it into a private function and call it from TEST_P(){}

Reply:
yeah this is a good catch)